src/pkg/clouds/gcp/storage.go

```diff
@@ -137,7 +140,26 @@ func (gcp Gcp) GetBucketObject(ctx context.Context, bucketName, objectName strin
 		return nil, fmt.Errorf("unable to get bucket object, failed to create storage client: %w", err)
 	}
 	defer client.Close()
+	return gcp.getBucketObject(ctx, bucketName, objectName, client)
+}
+
+func (gcp Gcp) GetBucketObjectWithServiceAccount(ctx context.Context, bucketName, objectName, serviceAccount string) ([]byte, error) {
+	ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+		TargetPrincipal: serviceAccount,
+		Scopes:          []string{"https://www.googleapis.com/auth/cloud-platform"},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get bucket object, failed to create impersonated token source for service account %v: %w", serviceAccount, err)
+	}
+	client, err := storage.NewClient(ctx, option.WithTokenSource(ts))
```

Comment:
Can we instantiate a single client and attach it to the `Gcp` struct? That way we can mock it and test this code?